### ISSUE TITLE
Error after jailbreaking -20 (very cummon issue)

### ISSUE THREAD

**Issue by nxtsite** [checkra1n/BugTracker#161]
POSSIBLE FIX : HOLD VOLUME UP BUTTON WHILE JAILBREAKING TO BOOT INTO SAFE MODE (NOT TRIED IT YET BUT MAYBE THIS WILL FIX FOR YOU, LET ME KNOW IF IT WORKS)

**What device + iOS version are you on?**
iOS 13.2.2

**What checkra1n version are you using?**
0.9.1 beta
**What are the steps to reproduce the issue?**
1. Run checkra1n jailbreak
2. Use phone and then reboot
3. Run checkra1n again and get the error
...

**What do you expect, and what is happening instead?**
I expect to run the jailbreak, but i get the error( not just me, all my friends)

**Any other info, error logs, screenshots, ...?**
Just search error -20 on twitter i dont have screenshot

**Comment by nxtsite** [comment]
![FD058CE1-0926-4719-A9D5-AF2620BAEA43](https://user-images.githubusercontent.com/56782850/68564583-b7c59080-0459-11ea-8359-bff35f405fb5.jpeg)
![1C0CCD23-CB7F-409E-8335-6285CDC9FA6A](https://user-images.githubusercontent.com/56782850/68564584-b85e2700-0459-11ea-90d5-10be31688078.jpeg)

**Comment by perpelator** [comment]
I also have this issue on the iPhone 10.

**Comment by cjf99** [comment]
same. iPhone X iOS 13.2.2

**Comment by kasperbeck** [comment]
Same here.  I figured out error -26 and -31, but this one just popped up for me while booting.
iPhone 8 ios 13.2.2 on a Macbook Air, High Sierra 10.13.6

Let me know if you guys figure anything out

**Comment by nxtsite** [comment]
@kasperbeck how did you fix error -26 & -31? Maybe i will need it

**Comment by cjf99** [comment]
I got it fixed! Just do everything in the program but when the dfu guide comes start at 5 seconds not at 6 at the first step. Everything else normally! Try it a few times for me it worked on the 2nd try. Shoutout to @opa334 for telling me on twitter!

**Comment by kasperbeck** [comment]
@nxtsite  I fixed the -26 and -31 error by switching out my usb chord for a different one and also by pluging it directly into my mac instead of into a multiple usb port accessory.
@cjf99 I'm about to try out your tip right now! wish me luck

**Comment by nxtsite** [comment]
@kasperbeck @cjf99 thank you, i will try in a few hours and tell you if this works

**Comment by nullpixel** [comment]
What tweaks did you install?

**Comment by kasperbeck** [comment]
@nxtsite hahah well not the errors are back, so I'm not sure again.  I'll keep you updated

**Comment by nxtsite** [comment]
@nullpixel oh HI , thanks for reaching. So i installed a tweak that modified unlock and lock sounds and after pressing Restart Springboard it showed that loading sign and then it stopped. !!It was not [[UNLOCKSOUND]] , it was another tweak for this but i forgot the name. Maybe you will remember it for me

**Comment by nxtsite** [comment]
@kasperbeck ok thanks

**Comment by nxtsite** [comment]
@nullpixel if i am right, the tweak name was[[SoundLock]]

**Comment by hackcrypto** [comment]
hello, i have iphone 5s and i have the same error and the boot with checkra1n work but the jailbreak is not installed 
![Photo le 11-11-2019 à 10 44 #2](https://user-images.githubusercontent.com/33383620/68578041-86f15580-0471-11ea-9591-ae810f5f2043.jpg)

**Comment by nxtsite** [comment]
@hackcrypto this error is specific to “-20” , just wait for the update to let you boot into safe mode and delete the bad tweak you installed. @qwertyoruiopz said that this update is coming ;)

**Comment by hackcrypto** [comment]
 ok, thank you for your answer @nxtsite 

**Comment by FraserEmbrey** [comment]
I have this issue on my iPhone X. No tweaks installed first time jailbreaking the phone

**Comment by nxtsite** [comment]
@FraserEmbrey maybe for you the problem is in the cable or in the usb port you use. Try changing it. If it does not work, wait for update ;)

**Comment by furby8704** [comment]
> I have this issue on my iPhone X. No tweaks installed first time jailbreaking the phone

I had this issue. if you're on macOS catalina you must manually add checkra1n in full disk access in privacy and settings to get it to run

**Comment by mehmeteminkartal** [comment]
I have the same issue. iPhone X(GSM, 13.2.2) with Checkra1n beta 0.9.1 using macOS 10.15.2. Tried in two devices and same issue. Tried giving checkra1n full disk access as @furby8704 pointed out, but still same error. Even tried running checkra1n as root. 
`Timed out waiting for bootstrap upload (Error -20)`
<img width="801" alt="image" src="https://user-images.githubusercontent.com/5096916/68586158-0e9c8b80-0495-11ea-8ee6-754e6ad3de31.png">
I've connected my iPhone directly to my MacBook without any hub
<img width="273" alt="image" src="https://user-images.githubusercontent.com/5096916/68586196-296f0000-0495-11ea-8b67-f8c0701ae73a.png">

**Comment by perpelator** [comment]
@cjf99 I used your method and it worked for me! Thanks!

**Comment by FraserEmbrey** [comment]
> > I have this issue on my iPhone X. No tweaks installed first time jailbreaking the phone
> 
> I had this issue. if you're on macOS catalina you must manually add checkra1n in full disk access in privacy and settings to get it to run

I have the same issue after allowing full disk access. I cannot get into DFU mode following the guide either. I have been hard resetting the iPhone. I have used Apple cables USB-C to lightning and the iPhone is running iOS 13.2.2, the MacBook Pro is a 2018 model running macOS 10.15.1

**Comment by nxtsite** [comment]
@cjf99 could you please explin more detailed your method of fixing so i can pin it ?

**Comment by perpelator** [comment]
@nxtsite I used the same method as him. Here’s it explained a bit better: 1. Plug in your device. 2: Click start and get into recovery mode via Checkra1n (don’t manually go into recovery mode via buttons yourself. Let checkra1n do it for you by pressing next). 3: Once your device is detected again, begin to start the DFU mode process. After get ready (which is 3 seconds), the next step will begin at 6 seconds. Do NOT begin until that hits 5 seconds. 4: Continue to do what it says at the normal amount of said time. 

Personally I started at 6 seconds but then cut all the other steps short by 1 second, but it’ll work either way. I personally was using a Hackintosh via usb a 2.0 on MacOS 10.13.6. Cannot confirm this method works (but it should) for other versions. I tested this on both 0.9 and 0.9.1, so should carry over with the following versions until fixed. :)
